### DISCUSSION TITLE
Massive Memory Usage When Rendering and Playing 6mb Lottie

### DISCUSSION THREAD

**Discussion by anishjain123** [airbnb/lottie-ios#2414] [General]
## Which Version of Lottie are you using?

Lottie 4.4.3

## Expected Behavior 
Low memory usage when Lottie is being rendered.
The Lottie file is 6.6MB (exported from AE with complex layering, then compressed with LottieFiles.com) and so I expect maybe 100MB of memory utilization tops. 

## Actual Behavior
![Screenshot 2024-05-24 at 3 04 15 PM](https://github.com/airbnb/lottie-ios/assets/5897312/35cdbc42-c0ad-4f1f-92ac-3c59ff4a4729)
When I load and play the Lottie in a very simple view with basically nothing else in it, the app memory usage is 1.2GB

When i exit that view and go back to the previous view (also very minimal) the app memory usage is 52MB, meaning over 1.15GB of memory is just the lottie. 

The only thing I could find online about this only was changing the rendering engine and I tried both Automatic and CoreAnimation and both yielded the same results.

This lottie (and 5-6 other ones I created, exported, and compressed in the same way) are CRITICAL to my application's UX and it would really mean the world if you could help me with this! 

Thanks for all the great work yall have done in creating and supporting this library, and thanks in advance for helping on this important issue :)

## Animation JSON
[Listening_lottie.json](https://github.com/airbnb/lottie-ios/files/15428383/Listening_lottie.json)

**Comment by anishjain123** [comment]
@calda @absolute-heike  I saw you both responded to similar issues so just want to tag you on this 😄

**Comment by anishjain123** [comment]
Is there any way we can maybe change the amount of frames that are buffered? I think that might be significantly impacting performance.

**Comment by calda** [comment]
Have you tried the Main Thread rendering engine?

A 6.6MB animation is very large so I’m not surprised that there are memory issues. Can the size of the animation be reduced? Are there embedded images, or just 6+ MB of json?

**Reply by anishjain123** [reply]
yep no change on main thread. 

There are embedded images so that's the brunt of the size. 

We tried many ways to decrease the size of the animation and this is about the best we can do!

Does the lottie buffer entirely into memory or just the next few frames/seconds? Is there any way to change how much of the lottie is being loaded into memory and offload some of that memory to more dynamic such as with video players? @calda

**Reply by calda** [reply]
The entire animation is loaded at once. If the animation is long, you could cut it into smaller pieces and play those sequentially.

How large are the embedded images? 6mb of images still sounds like a lot to me. Perhaps the size of those could be reduced.
